Sources/GoTrue/GoTrueKeychain.swift

```diff
@@ -0,0 +1,119 @@
+import Foundation
+import Security
+
+class GoTrueKeychain {
+    // Parameters for storage, set with sensible defaults in `init`
+    var serviceName: String
+    var accessGroup: String?
+    
+    // Private constants
+    private let key: Data = "supabase_session_key".data(using: .utf8)!
+    private let encoder = JSONEncoder()
+    private let decoder = JSONDecoder()
+    
+    init(serviceName: String? = nil, accessGroup: String? = nil) {
+        self.serviceName = serviceName ?? "supabase.gotrue.swift"
+        self.accessGroup = accessGroup
+    }
+    
+    func storeSession(_ session: Session) throws {
+        let data = try encoder.encode(session)
+        storeData(data)
+    }
+    
+    func getSession() throws -> Session? {
+        if let encodedData = getData() {
+            return try decoder.decode(Session.self, from: encodedData)
+        } else {
+            return nil
+        }
```

Comment:
Can be simplified by using a `map`.
```suggestion
        try getData().map {
            try decoder.decode(Session.self, from: $0)
        }
```

Reply:
Yeah that looks more elegant, thanks!